common/predictive-text/worker/correction/distance-modeler.ts

```diff
@@ -276,14 +286,15 @@ namespace correction {
     // Signals that the edge has already been processed.
     private processedEdgeSet: {[mapKey: string]: boolean} = {};
 
-    constructor(traversalRoot: LexiconTraversal) {
+    constructor(model: LexicalModel) {
+      if(!model || !model.traverseFromRoot) {
+        throw "The provided model does not meet the requirements needed to support robust correction searching.";
```

Comment:
What happens in KeymanWeb when a model that does not support `model.traverseFromRoot` gets passed to the distance modler?

Reply:
1.  If a model doesn't support `traverseFromRoot`, the context-modeler module is not instantiated.

https://github.com/keymanapp/keyman/blob/6a2c127ee6aeab23555f5f2aee9152c10e95bc29/common/predictive-text/worker/model-compositor.ts#L10-L14

Now, within `predict()`...

2. If the context-moduler module is not instantiated, we use the 'old' correction method:

https://github.com/keymanapp/keyman/blob/6a2c127ee6aeab23555f5f2aee9152c10e95bc29/common/predictive-text/worker/model-compositor.ts#L112-L118

3.  If it is instantiated, we use the 'new' stuff.

https://github.com/keymanapp/keyman/blob/6a2c127ee6aeab23555f5f2aee9152c10e95bc29/common/predictive-text/worker/model-compositor.ts#L137-L143

Note that `SearchSpace`, the distance-modeler's core, only exists on states returned by the context-modeler module.

https://github.com/keymanapp/keyman/blob/6a2c127ee6aeab23555f5f2aee9152c10e95bc29/common/predictive-text/worker/model-compositor.ts#L152-L158

Reply:
Excellent description!